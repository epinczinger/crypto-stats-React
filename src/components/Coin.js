import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Coin = ({
  id, symbol, name, price, change,
}) => {
  const theme = useSelector((state) => state.toggle);

  return (
    <div
      id={id}
      className={`${theme} border shadow mx-auto sm:w-full text-center w-7/8 flex flex-col sm:flex-row justify-between`}
    >
      <div className="p-2 sm:w-1/5">
        <p>{symbol}</p>
      </div>
      <div className="p-2 sm:w-1/5">
        <p>{name}</p>
      </div>
      <div className="p-2 sm:w-1/5">
        <p className="hidden sm:block">{price}</p>
        <p className="sm:hidden">{`usd ${price}`}</p>
      </div>
      <div className="p-2 sm:w-1/5">
        <p>{`${change} %`}</p>
      </div>
      <div className="p-2 sm:w-1/5">
        <Link to={`/coin/${id}`}>
          + Info
        </Link>
      </div>
    </div>
  );
};

export default Coin;

Coin.defaultProps = {
  id: '',
  symbol: '',
  name: '',
  price: '',
  change: '',
};

Coin.propTypes = {
  id: PropTypes.string,
  symbol: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.string,
  change: PropTypes.string,
};
