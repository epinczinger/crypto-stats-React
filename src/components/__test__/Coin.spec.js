import React from 'react';
import renderer from 'react-test-renderer';
import Coin from '../Coin';

describe('The Coin component', () => {
  it('renders as expected matching snapshot', () => {
    const tree = renderer.create(<Coin />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
