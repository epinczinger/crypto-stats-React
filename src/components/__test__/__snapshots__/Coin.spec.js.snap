// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Coin component renders as expected matching snapshot 1`] = `
<div
  className="border bg-gray-200 dark:bg-gray-800  hover:border-yellow-600 hover:bg-gray-300 dark:hover:bg-gray-700 shadow mx-auto sm:w-full text-center w-7/8 flex flex-col sm:flex-row justify-between"
  id=""
>
  <div
    className="p-2 text-lg sm:w-1/5"
  >
    <a
      href="/coin/"
      onClick={[Function]}
    >
      
    </a>
  </div>
  <div
    className="p-2 sm:w-1/5 text-xl"
  >
    <p>
      
    </p>
  </div>
  <div
    className="p-2 sm:w-1/5"
  >
    <p
      className="hidden sm:block"
    >
      $ 
    </p>
    <p
      className="sm:hidden text-xl"
    >
      $ 
    </p>
  </div>
  <div
    className="text-red-700 p-2 font-semibold sm:w-1/5"
  >
    <p>
       %
    </p>
  </div>
  <div
    className="p-2 sm:w-1/5 text-center"
  >
    <a
      href="/coin/"
      onClick={[Function]}
    >
      <p
        className="rounded-full mx-auto h-7 w-7 hover:text-gray-800 text-semibold border border-gray-600 flex items-center justify-center"
      >
        +
      </p>
    </a>
  </div>
</div>
`;
